Respond to feedback after 262943@main

achristensen07 opened the pull request:
#### 3ea88cae0eed9608777523ae826204f0ba7e2422
<pre>
Respond to feedback after 262943@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=255437">https://bugs.webkit.org/show_bug.cgi?id=255437</a>
rdar://108032616

Reviewed by Chris Dumez.

Chris gave some excellent feedback.  This takes it into account to polish the code a little more.
I also move the maps to WebPageProxy::Internals so we don&apos;t have to include an extra header from
WebPageProxy.h.

* Source/WebKit/UIProcess/ProvisionalFrameProxy.cpp:
(WebKit::ProvisionalFrameProxy::ProvisionalFrameProxy):
* Source/WebKit/UIProcess/ProvisionalFrameProxy.h:
* Source/WebKit/UIProcess/SubframePageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::subframePageProxyForFrameID const):
(WebKit::WebPageProxy::addSubframePageProxyForFrameID):
(WebKit::WebPageProxy::registrableDomainForSubframeID const):
(WebKit::WebPageProxy::subpageFrameProxyForRegistrableDomain const):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::loadRequestByCreatingNewLocalFrameOrConvertingRemoteFrame):

Canonical link: <a href="https://commits.webkit.org/262973@main">https://commits.webkit.org/262973@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/02c85eb6300a15fd16456228379456e90fd5872f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3170 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3231 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3341 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4583 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3528 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3302 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3258 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2758 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3201 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/3526 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2849 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4398 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/1025 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2822 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/2670 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/2804 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2873 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/4147 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3246 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/2592 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2830 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2824 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2823 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/371 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3089 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/UIProcess/ProvisionalFrameProxy.cpp:
```diff
@@ -49,7 +49,7 @@
 namespace WebKit {
 
-ProvisionalFrameProxy::ProvisionalFrameProxy(WebFrameProxy& frame, Ref<WebProcessProxy>&& process, const WebCore::ResourceRequest& request, bool newProcess)
+ProvisionalFrameProxy::ProvisionalFrameProxy(WebFrameProxy& frame, Ref<WebProcessProxy>&& process, const WebCore::ResourceRequest& request, bool shouldUseNewProcess)
```
I know I suggested this name but after I read the rest of the patch, I realized that my understanding was likely incorrect. This isn't a `shouldUseNewProcess` but rather a `isNewProcess` I believe.

cdumez replied:
or didCreateNewProcess.

cdumez commented on Source/WebKit/UIProcess/ProvisionalFrameProxy.cpp:
```diff
@@ -87,7 +87,7 @@ ProvisionalFrameProxy::ProvisionalFrameProxy(WebFrameProxy& frame, Ref<WebProces
     auto giveAllCookieAccess = LoadedWebArchive::Yes;
-    if (newProcess) {
+    if (shouldUseNewProcess) {
         page.websiteDataStore().networkProcess().sendWithAsyncReply(Messages::NetworkProcess::AddAllowedFirstPartyForCookies(m_process->coreProcessIdentifier(), WebCore::RegistrableDomain(request.url()), giveAllCookieAccess), [process = m_process, loadParameters = WTFMove(loadParameters), pageID = m_pageID] () mutable {
```
Why the blank line here?

achristensen07 replied:
What blank line?

cdumez replied:
I don't see one in the source. I guess I got confused by the way Github presented the diff. It looked like there was a blank line after the if check.

cdumez commented on Source/WebKit/UIProcess/WebPageProxyInternals.h:
```diff
@@ -212,6 +213,9 @@ struct WebPageProxy::Internals final : WebPopupMenuProxy::Client
     WebCore::PageIdentifier webPageID;
     WindowKind windowKind { WindowKind::Unparented };
 
+    HashMap<WebCore::RegistrableDomain, UniqueRef<SubframePageProxy>>  domainToSubframePageProxyMap;
```
Sill too many spaces between type and name

cdumez commented on Source/WebKit/UIProcess/WebPageProxyInternals.h:
```diff
@@ -212,6 +213,9 @@ struct WebPageProxy::Internals final : WebPopupMenuProxy::Client
     WindowKind windowKind { WindowKind::Unparented };
 
+    HashMap<WebCore::RegistrableDomain, UniqueRef<SubframePageProxy>>  domainToSubframePageProxyMap;
+    HashMap<WebCore::FrameIdentifier, WebCore::RegistrableDomain>  registrableDomainForSubframeID;
```
Same spacing issue and still the old name that is not great.

cdumez reviewed (approved):
LGTM

achristensen07 committed:
Respond to feedback after 262943@main
https://bugs.webkit.org/show_bug.cgi?id=255437
rdar://108032616

Reviewed by Chris Dumez.

Chris gave some excellent feedback.  This takes it into account to polish the code a little more.
I also move the maps to WebPageProxy::Internals so we don't have to include an extra header from
WebPageProxy.h.

* Source/WebKit/UIProcess/ProvisionalFrameProxy.cpp:
(WebKit::ProvisionalFrameProxy::ProvisionalFrameProxy):
* Source/WebKit/UIProcess/ProvisionalFrameProxy.h:
* Source/WebKit/UIProcess/SubframePageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::subframePageProxyForFrameID const):
(WebKit::WebPageProxy::addSubframePageProxyForFrameID):
(WebKit::WebPageProxy::registrableDomainForSubframeID const):
(WebKit::WebPageProxy::subpageFrameProxyForRegistrableDomain const):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::loadRequestByCreatingNewLocalFrameOrConvertingRemoteFrame):

Canonical link: https://commits.webkit.org/262973@main
